nrespipe/utils.py

```diff
@@ -651,3 +651,44 @@ def ingest_file(file_path):
                          extra={'tags': {'filename': file_path}})
             retry = True
             try_counter += 1
+
+
+def need_to_process(file_info, db_address):
+    should_process = True
+
+    # Note if this is an archived_fits message it has the whole header already attached
+    if file_info.get('version_set') is not None and not is_raw_nres_file(file_info):
+        should_process = False
+
+    path, filename, checksum = get_path_info(file_info)
+    if filename_is_blacklisted(filename):
+        logger.debug('Filename does not pass black list. Skipping...', extra={'tags': {'filename': filename}})
+        should_process = False
+
+    if not already_processed(filename, checksum, db_address):
+        logger.debug('NRES File already processed. Skipping...', extra={'tags': {'filename': filename}})
+        should_process = False
+
+    if not should_process:
+        dbs.set_file_as_processed(filename, checksum, file_info.get('frameid'), db_address)
```

Comment:
Just checking, this isn't the datastore that you said could only handle one update at a time right?